components/server/src/auth/host-context-provider-impl.ts

```diff
@@ -28,17 +28,30 @@ export class HostContextProviderImpl implements HostContextProvider {
     @inject(HostContextProviderFactory)
     protected factory: HostContextProviderFactory;
 
+    async init() {
+        await this.ensureInitialized();
+    }
+
     protected initialized = false;
-    protected ensureInitialized() {
+    protected async ensureInitialized() {
         if (this.initialized) {
             return;
         }
-        this.createFixedHosts()
-        this.updateDynamicHosts();
+        this.createFixedHosts();
+
+        try {
+            await this.updateDynamicHosts();
+        } catch (error) {
+            log.error(`Failed to update dynamic hosts.`, error);
+        }
 
         // schedule periodic update of dynamic hosts
         const scheduler = () => setTimeout(async () => {
```

Comment:
can the update of dynamic hosts take longer than two seconds? If so, what would happen?

Reply:
AFAIK in cases like those we usually resort a chained `setTimeout` variant.